tools/smoke_test.py

```diff
@@ -0,0 +1,174 @@
+import os
+import re
+import subprocess
+import sys
+import traceback
+
+try:
+    import packaging.version
+except ImportError:
+    from pkg_resources import packaging
+Version = packaging.version.Version
+
+MIN_CUDA_VERSION = Version("11.6")
+MIN_PYTHON_VERSION = (3, 7)
+# NOTE: dev version required, e.g. Version('1.13.0.dev123') < Version('1.13.0')
+MIN_TORCH_VERSION = Version("1.13.0.dev0")
+
+
+def check_python():
+    if sys.version_info < MIN_PYTHON_VERSION:
+        raise RuntimeError(
+            f"Python version not supported: {sys.version_info} "
+            f"- minimum requirement: {MIN_PYTHON_VERSION}"
+        )
+
+
+# Checks for correct pip dependencies, according to `install_requires` in setup.py.
+def check_pip_deps():
+    proc = subprocess.run(["pip", "check"], capture_output=True)
+    if proc.returncode != 0:
+        raise RuntimeError("`pip` dependencies broken:\n" + proc.stdout.decode("utf-8"))
+    proc = subprocess.run(["pip", "show", "torchdynamo"], capture_output=True)
+    if proc.returncode != 0:
+        raise RuntimeError("`torchdynamo` is not installed")
+
+
+# Checks for correct torch version.
+# Using check_pip_deps does not work if the minimum required torch version
+# is not present in PyPI.
+def check_torch():
+    import torch
+
+    if torch.__version__ < MIN_TORCH_VERSION:
+        raise RuntimeError(
+            f"torch version not supported: {torch.__version__} "
+            f"- minimum requirement: {MIN_TORCH_VERSION}"
+        )
+
+
+def check_cuda():
+    import torch
+
+    if not torch.cuda.is_available():
+        sys.stderr.write("CUDA not available for torch\n")
```

Comment:
It's not error, if users want try torchdynamo on CPU, CUDA is not necessary. Just print out if CUDA is available and version is supported.